lib/ClangImporter/ImportName.cpp

```diff
@@ -1407,6 +1407,20 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
   ArrayRef<const clang::ParmVarDecl *> params;
   switch (D->getDeclName().getNameKind()) {
   case clang::DeclarationName::CXXConstructorName:
+    isInitializer = true;
+    isFunction = true;
+    result.info.initKind = CtorInitializerKind::Designated;
+    baseName = "init";
+    // Add empty argument names.
+    if (auto ctor = dyn_cast<clang::CXXConstructorDecl>(D)) {
+      for (size_t i = 0; i < ctor->param_size(); ++i) {
+        argumentNames.push_back(StringRef());
+      }
+      if (ctor->isVariadic())
+        argumentNames.push_back(StringRef());
+    }
```

Comment:
Maybe don't duplicate this code. This is identical for functions and it will be the same for operators.

Reply:
Pulled it out into a separate function. This has also allowed me to eliminate the comment, as it now just duplicates what the code says anyway.